modules/examples/sources/msg-generator/msg-generator.c

```diff
@@ -74,7 +74,8 @@ static void
 msg_generator_sd_free(LogPipe *s)
 {
   MsgGeneratorSourceDriver *self = (MsgGeneratorSourceDriver *) s;
-
+  g_hash_table_remove_all(self->source_options.name_value);
```

Comment:
You do not need to remove all elements here. `g_hash_table_unref` below will do so.